### PR TITLE
IMPROVE storybook's compatibility to work within a monorepo

### PR DESCRIPTION
**Pull request by ndelangen**
Issue: https://github.com/storybookjs/storybook/issues/3346 https://github.com/storybookjs/storybook/issues/5949

## What I did
- Change the exclude to a regex, this helps with monorepo support, because in monorepos the `node_modules` folders might be more then 1, and not in the same place always. making this a regex makes sure that we never transform something from a path that has `node_modules in it`.
- Change the includes filter. In a monorepo storybook might be ran from a subdir in a monorepo, but files relevant for storybook might be higher up the tree. So I try to find the project root and include that.
- Add sourceType=unumbiguous to babel options so it can load commonjs modules.

### CODE REVIEW

File: lib/core/src/server/config/utils.js [hunk 1]
```diff
@@ -3,7 +3,7 @@ import { getEnvironment } from 'lazy-universal-dotenv';
 
 export const includePaths = [path.resolve('./')];
 export const nodeModulesPaths = path.resolve('./node_modules');
```
Comment (shilman) [inline lib/core/src/server/config/utils.js]: Unused var

Comment (ndelangen) [reply lib/core/src/server/config/utils.js]: it's exported...

Comment (shilman) [reply lib/core/src/server/config/utils.js]: 🙈

Comment (ndelangen) [reply lib/core/src/server/config/utils.js]: 😘

File: lib/core/src/server/config/utils.js [hunk 2]
```diff
@@ -1,9 +1,31 @@
...
+    //
+  }
+
+  return result || process.cwd();
+};
+
+export const includePaths = [projectRoot()];
 export const nodeModulesPaths = path.resolve('./node_modules');
-export const excludePaths = [nodeModulesPaths];
+export const excludePaths = [/node_modules/];
```
Comment (hipstersmoothie) [inline lib/core/src/server/config/utils.js]: I can confirm that this line fixes my problems in a monorepo 🎉

**Comment by astrotim** [comment]
Is the `next-6.0.0` branch available via npm?

**Comment by ndelangen** [comment]
no not yet

**Comment by shilman** [comment]
@astrotim Nope. I plan to release 5.3 this weekend and then cut over so that 6.0 gets released on the `next` NPM tag. Coming soon!

**Comment by astrotim** [comment]
OK great, I'm looking forward to taking it for a spin in our monorepo

**Comment by lucasleelz** [comment]
Is there an example of how to use it? I have upgraded to `6.0.0-alpha.31`

**Pull request merged**